Common/Tools/Multiplicity/multGlauberNBDFitter.cxx

```diff
@@ -0,0 +1,393 @@
+
+  //master function
+  fGlauberNBD = new TF1("fGlauberNBD", this, &multGlauberNBDFitter::ProbDistrib,
+                        0, 50000, 4, "multGlauberNBDFitter", "ProbDistrib");
+  fGlauberNBD->SetParameter(0, fMu);
+  fGlauberNBD->SetParameter(1, fk);
+  fGlauberNBD->SetParameter(2, ff);
+  fGlauberNBD->SetParameter(3, fnorm);
+}
+
+multGlauberNBDFitter::multGlauberNBDFitter(const char* name, const char* title) : TNamed(name, title),
+                                                                                  fNBD(0x0),
+                                                                                  fhNanc(0x0),
+                                                                                  fhNpNc(0x0),
+                                                                                  ffChanged(kTRUE),
+                                                                                  fCurrentf(-1),
+                                                                                  fAncestorMode(2),
+                                                                                  fNpart(0x0),
+                                                                                  fNcoll(0x0),
+                                                                                  fContent(0x0),
+                                                                                  fNNpNcPairs(-1),
+                                                                                  fMaxNpNcPairs(1000000),
+                                                                                  fMu(45),
+                                                                                  fk(1.5),
+                                                                                  ff(0.8),
+                                                                                  fnorm(100),
+                                                                                  fFitOptions("R0"),
+                                                                                  fFitNpx(5000)
+{
+  //Named constructor
+  fNpart = new Double_t[fMaxNpNcPairs];
+  fNcoll = new Double_t[fMaxNpNcPairs];
+  fContent = new Long_t[fMaxNpNcPairs];
+
+  //Ancestor histo
+  //fhNanc = new TH1D("fhNanc", "", fAncestorMode==2?10000:1000, -0.5, 999.5);
+
+  //NBD
+  fNBD = new TF1("fNBD", "ROOT::Math::negative_binomial_pdf(x,[0],[1])", 0, 45000);
+  fNBD->SetNpx(45000);
+
+  //master function
+  fGlauberNBD = new TF1("fGlauberNBD", this, &multGlauberNBDFitter::ProbDistrib,
+                        0, 50000, 4, "multGlauberNBDFitter", "ProbDistrib");
+  fGlauberNBD->SetParameter(0, fMu);
+  fGlauberNBD->SetParameter(1, fk);
+  fGlauberNBD->SetParameter(2, ff);
+  fGlauberNBD->SetParameter(3, fnorm);
+}
+//________________________________________________________________
+multGlauberNBDFitter::~multGlauberNBDFitter()
+{
+  // Destructor
+  if (fNBD) {
+    delete fNBD;
+    fNBD = 0x0;
```

Comment:
If you allocate with `[]` you have to use `delete [] fNBD`
The same for the other arrays

Reply:
OK!
My fault! You did it correctly for the array cases!
Sorry!